### ISSUE TITLE
Not receiving notifications

### ISSUE THREAD

**Issue by mullhol** [StatCan/WellbeingCheckUAT#225]
I updated to the new version of the app yesterday around 11:30. I revised my notification window, and completed a questionnaire B. I have not received any notifications, and my notification schedule has not updated (it reflects a start date of October 13). I completed another questionnaire B this morning but my notification schedule still has not updated.

**Comment by AJBouzidi** [comment]
@mullhol could you please change the notification numbers?

**Comment by mullhol** [comment]
@AJBouzidi 

I just revised to 3 notifications from 2. I completed another questionnaire B as well.  The notification schedule has not updated.  I will let you know if I receive a notification before the end of day.

**Comment by mullhol** [comment]
Closed in error. Reopened.

**Comment by mullhol** [comment]
@AJBouzidi no change, still no notifications

**Comment by AJBouzidi** [comment]
@mullhol Could you remove the app and reinstall again.

**Comment by mullhol** [comment]
@AJBouzidi I'll try that tonight, but if it works, that brings up the previous issue we've discussed of what if we deploy an update during production?  This can't only work on first download.

**Comment by mullhol** [comment]
Further breakdown of the activity: 

I downloaded app version 1.4.6 when it was available. I completed survey submissions for a day or two. I then ignored the app for 4 days until I received the last chance notification. I ignored this notification as well. 

I updated to version 1.4.7 when it was made available. I changed my notification window. I completed a survey B. I checked the info button, and the schedule still reflected the week of October 13. I went back to the settings menu and changed the number of notifications. I completed another survey B.  I checked the info button, but the schedule still reflected the week of October 13.  I restarted my phone, then completed another survey B. The schedule still reflected the week of October 13.  I did not receive any notifications between the day I updated and Thursday evening.  I deleted the app on Thursday evening and installed a new version.  I customized my preferences in the settings menu, and completed a survey.  I now have a notification schedule and received my first notification this morning on time.  This implies issue 204 is still valid - there is a problem when we UPDATE the app from one version to the other.

**Comment by Jiandong-Guo** [comment]
there are 2 factors to prevent app from creating new schedule list:
   1) global.notificationState flag,  since the number of notification can be changed, it still is true after upgrade.
   2) OS Permissions.NOTIFICATIONS,  it was asked when the app was initially installed, maybe after upgrade, it is reset to default false.  will add call to askPermission again after upgrade to test.    this OS setting is true by default for android,  Is this problem happened to android too ?

**Comment by mullhol** [comment]
@Jiandong-Guo At this time we don't know if it's true for android, I've been having them delete the app before installing the new version rather than update.  I will ask them to update only for the next release so we know if they have an issue as well.

**Comment by mullhol** [comment]
I tested the same scenario as before - I received the last chance notification, ignored the app another 24 hours, then updated app version.  When I logged in, I checked the info button.  No schedule was present.  I completed a questionnaire B, and checked the info button again.  There is a schedule and the number of notifications per day are correct.  No days are missing in the schedule either. I received my first notification as scheduled at 10:51.
